### PR TITLE
add support for spamhaus DQS

### LINKED ISSUE #5: Suggestion: add a setting for Spamhaus key to re-enable use from public DNS
Not an issue but a enhancement suggestion. Spamhaus now rejects queries from public DNS resolvers by default (https://github.com/haraka/haraka-plugin-uribl/pull/3), [except when the query uses a key](https://www.spamhaus.com/resource-center/successfully-accessing-spamhauss-free-block-lists-using-a-public-dns/) (and complies to Spamhaus's fair use policy), which can be obtained for free from Spamhaus.

It'd be nice to have a setting to set that Spamhaus key in the plugin, thus re-enabling use of Spamhaus from public DNS resolvers.

### PR DESCRIPTION
**Pull request by DoobleD**
Fixes #5. Tested using https://blt.spamhaus.com/.

Changes proposed in this pull request:
- add a `dqs_key` config option
- add a `[dbl.dq.spamhaus.net]` zone, commented by default

Checklist:
- [x] docs updated
- [ ] tests updated
- [x] Changes.md updated
- [x] package.json.version bumped

### CODE REVIEW

**Comment by msimerson** [comment]
Thanks @DoobleD , this looks good. Can you add a suggested instructions for using this URIBL, either in the README or in comments in the config file?

**Comment by msimerson** [comment]
Instead of duplicating the entire spamhaus block, take a look at #8 and consider something like that, which has inline instructions on how to use one or the other.

**Comment by DoobleD** [comment]
Thanks for the feedback @msimerson.

Your config approach is better, I've copied it. I also added instructions in the README, though they may be somewhat redundant with the comments in the config file. I'll let you be the judge.

File: README.md [hunk 1]
```diff
@@ -85,6 +85,21 @@ The following are optional for each list:
 
   Specifies that the list requires hostnames be stripped down to the domain boundaries prior to querying the list.  This is required for the [SURBL][3] and [URIBL][4] lists.
 
+* dqs_key
```
Comment (msimerson) [inline README.md:88]: this is a SpamHaus specific setting. Should it be under the Spamhaus DQS heading?

**Comment by msimerson** [comment]
> though they may be somewhat redundant

You are right. Best to put them in one place or the other. When they're in more than one location, one set of the instructions tends to get stale. I lean towards keeping them in the config file close to the point of use but you can decide which is best.

File: README.md (hunk 1 above)
Comment (DoobleD) [reply README.md]: Sounds reasonable, I've moved it there.

**Comment by DoobleD** [comment]
> > though they may be somewhat redundant
> 
> You are right. Best to put them in one place or the other. When they're in more than one location, one set of the instructions tends to get stale. I lean towards keeping them in the config file close to the point of use but you can decide which is best.

Sounds good, I've removed the bit from the README.

**Comment by msimerson** [comment]
This looks great. Are you happy with it?

**Review by msimerson — APPROVED** [review]

**Comment by DoobleD** [comment]
> This looks great. Are you happy with it?

Yep, good to merge for me. Thank you! :)

**Pull request merged**